src/analyzer/protocol/websocket/unmask.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2024 by the Zeek Project. See COPYING for details.
+
+// Import HILTI
+#include <hilti/rt/libhilti.h>
+
+namespace hlt_websocket::WebSocket {
+
+// Implement XOR unmasking of WebSocket frames due to being very
+// hot and inefficient in pure Spicy.
+//
+// https://github.com/zeek/spicy/issues/1663
+::hilti::rt::Bytes fast_unmask(const hilti::rt::integer::safe<uint64_t>& masking_key_idx,
+                               const hilti::rt::Vector<hilti::rt::integer::safe<uint8_t>>& masking_key,
+                               const hilti::rt::Bytes& chunk) {
+    constexpr size_t masking_key_size = 4;
+
+    if ( masking_key.size() != masking_key_size )
+        throw hilti::rt::RuntimeError(hilti::rt::fmt("wrong masking_key size %ld", masking_key.size()));
```

Comment:
`RuntimeError` is just a base class and we probably have a type fitting better.

- if this is due to some error in the input and you want to recover from it use a `RecoverableFailure`
- if this is due to a misuse of this function use a `UsageError`